tests/test_georaster.py

```diff
@@ -409,48 +411,147 @@ def test_masking(self, example: str) -> None:
+
+        # bottom
+        cropGeom2 = [cropGeom[0], cropGeom[1] + rand_float * abs(r.res[1]), cropGeom[2], cropGeom[3]]
+        r_cropped = r.crop(cropGeom2, inplace=False)
+        assert r.shape[0] - (r_cropped.bounds.top - r_cropped.bounds.bottom) / r.res[1] == int(rand_float)
+        assert gu.misc.array_equal(r.data[:, : -int(rand_float), :], r_cropped.data)
+
+        # top
+        cropGeom2 = [cropGeom[0], cropGeom[1], cropGeom[2], cropGeom[3] - rand_float * abs(r.res[1])]
+        r_cropped = r.crop(cropGeom2, inplace=False)
+        assert r.shape[0] - (r_cropped.bounds.top - r_cropped.bounds.bottom) / r.res[1] == int(rand_float)
+        assert gu.misc.array_equal(r.data[:, int(rand_float) :, :], r_cropped.data)
+
+        # -- Test with mode='match_extent' -- #
+        # Test all sides at once, with rand_float less than half the smallest extent
+        # The cropped extent should exactly match the requested extent, res will be changed accordingly
+        rand_float = np.random.randint(1, min(r.shape) / 2 - 1) + 0.25
+        cropGeom2 = [
+            cropGeom[0] + rand_float * r.res[0],
+            cropGeom[1] + rand_float * abs(r.res[1]),
+            cropGeom[2] - rand_float * r.res[0],
+            cropGeom[3] - rand_float * abs(r.res[1]),
+        ]
+        r_cropped = r.crop(cropGeom2, inplace=False, mode="match_extent")
+        assert list(r_cropped.bounds) == cropGeom2
+        # The change in resolution should be less than what would occur with +/- 1 pixel
+        assert np.all(
+            abs(np.array(r.res) - np.array(r_cropped.res)) < np.array(r.res) / np.array(r_cropped.shape)[::-1]
+        )
 
-        r = gr.Raster(self.landsat_b4_path)
-        r2 = gr.Raster(self.landsat_b4_crop_path)
+        r_cropped2 = r.crop(r_cropped, inplace=False, mode="match_extent")
+        assert r_cropped2.bounds == r_cropped.bounds
+        assert gu.misc.array_equal(r_cropped2.data, r_cropped.data)
 
-        # Read a vector and extract only the largest outline within the extent of r
-        outlines = gu.Vector(examples.get_path("everest_rgi_outlines"))
+        # -- Test with CropGeom being a Vector -- #
+        outlines = gu.Vector(outlines_path)
         outlines.ds = outlines.ds.to_crs(r.crs)
-        outlines.crop2raster(r)
-        outlines = outlines.query(f"index == {np.argmax(outlines.ds.geometry.area)}")
+        # outlines.crop2raster(r)
+        # outlines = outlines.query(f"index == {np.argmax(outlines.ds.geometry.area)}")
+        # Crop and validate that it got smaller
+        r_cropped = r.crop(outlines, inplace=False)
+        assert r.data.size > r_cropped.data.size  # type: ignore
 
-        # Crop the raster to the outline and validate that it got smaller
-        r_outline_cropped = r.crop(outlines, inplace=False)
-        assert r.data.size > r_outline_cropped.data.size  # type: ignore
+        # b = r.bounds
+        # b2 = r2.bounds
```

Comment:
The problem is that it will not necessarily match, because of differences in resolution, and because the raster's extent might be smaller than the outlines extent.
I will try to find a way to check that using intersection.

Reply:
Solved with 9d22fc9